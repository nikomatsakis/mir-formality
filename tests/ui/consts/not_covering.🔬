[
    crate Foo {
        trait Foo<const C> where type_of_const C is bool {}

        trait Bar<ty T, const C> where T: Foo<const C>, type_of_const C is bool {}

        impl<> Foo<const true> for u32 {}
        impl<const C> Bar<u32, const C> for u32 where type_of_const C is bool {}
    }
]
