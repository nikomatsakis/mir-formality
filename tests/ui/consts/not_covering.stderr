Error: check_trait_impl(impl <const> Bar <u32, const ^const0_0> for u32 where type_of_const ^const0_0 is bool { })

Caused by:
    judgment `prove_wc_list { goal: {Bar(u32, u32, const !const_0)}, assumptions: {@ ConstHasType(!const_0 , bool)}, env: Env { variables: [!const_0], coherence_mode: false }, decls: decls(222, [trait Foo <ty, const> where {@ ConstHasType(^const0_1 , bool)}, trait Bar <ty, ty, const> where {Foo(^ty0_1, const ^const0_2), @ ConstHasType(^const0_2 , bool)}], [impl Foo(u32, const value(0, bool)), impl <const> Bar(u32, u32, const ^const0_0) where {@ ConstHasType(^const0_0 , bool)}], [], [], [], [], {Bar, Foo}, {}) }` failed at the following rule(s):
      the rule "some" failed at step #0 (crates/formality-prove/src/prove/prove_wc_list.rs:28:14) because
        judgment `prove_wc { goal: Bar(u32, u32, const !const_0), assumptions: {@ ConstHasType(!const_0 , bool)}, env: Env { variables: [!const_0], coherence_mode: false }, decls: decls(222, [trait Foo <ty, const> where {@ ConstHasType(^const0_1 , bool)}, trait Bar <ty, ty, const> where {Foo(^ty0_1, const ^const0_2), @ ConstHasType(^const0_2 , bool)}], [impl Foo(u32, const value(0, bool)), impl <const> Bar(u32, u32, const ^const0_0) where {@ ConstHasType(^const0_0 , bool)}], [], [], [], [], {Bar, Foo}, {}) }` failed at the following rule(s):
          the rule "exhaustive positive impl" failed at step #3 (crates/formality-prove/src/prove/prove_wc.rs:68:14) because
            judgment `collect` failed at the following rule(s):
              failed at (crates/formality-core/src/judgment/proven_set.rs:192:13) because
                empty collection
          the rule "positive impl" failed at step #7 (crates/formality-prove/src/prove/prove_wc.rs:100:14) because
            judgment `prove_after { constraints: Constraints { env: Env { variables: [!const_0, ?const_1], coherence_mode: false }, known_true: true, substitution: {?const_1 => const !const_0} }, goal: {Foo(u32, const ?const_1), @ ConstHasType(?const_1 , bool)}, assumptions: {@ ConstHasType(!const_0 , bool)}, decls: decls(222, [trait Foo <ty, const> where {@ ConstHasType(^const0_1 , bool)}, trait Bar <ty, ty, const> where {Foo(^ty0_1, const ^const0_2), @ ConstHasType(^const0_2 , bool)}], [impl Foo(u32, const value(0, bool)), impl <const> Bar(u32, u32, const ^const0_0) where {@ ConstHasType(^const0_0 , bool)}], [], [], [], [], {Bar, Foo}, {}) }` failed at the following rule(s):
              the rule "prove_after" failed at step #1 (crates/formality-prove/src/prove/prove_after.rs:19:14) because
                judgment `prove_wc_list { goal: {Foo(u32, const !const_0), @ ConstHasType(!const_0 , bool)}, assumptions: {@ ConstHasType(!const_0 , bool)}, env: Env { variables: [!const_0], coherence_mode: false }, decls: decls(222, [trait Foo <ty, const> where {@ ConstHasType(^const0_1 , bool)}, trait Bar <ty, ty, const> where {Foo(^ty0_1, const ^const0_2), @ ConstHasType(^const0_2 , bool)}], [impl Foo(u32, const value(0, bool)), impl <const> Bar(u32, u32, const ^const0_0) where {@ ConstHasType(^const0_0 , bool)}], [], [], [], [], {Bar, Foo}, {}) }` failed at the following rule(s):
                  the rule "some" failed at step #0 (crates/formality-prove/src/prove/prove_wc_list.rs:28:14) because
                    judgment `prove_wc { goal: Foo(u32, const !const_0), assumptions: {@ ConstHasType(!const_0 , bool)}, env: Env { variables: [!const_0], coherence_mode: false }, decls: decls(222, [trait Foo <ty, const> where {@ ConstHasType(^const0_1 , bool)}, trait Bar <ty, ty, const> where {Foo(^ty0_1, const ^const0_2), @ ConstHasType(^const0_2 , bool)}], [impl Foo(u32, const value(0, bool)), impl <const> Bar(u32, u32, const ^const0_0) where {@ ConstHasType(^const0_0 , bool)}], [], [], [], [], {Bar, Foo}, {}) }` failed at the following rule(s):
                      the rule "exhaustive positive impl" failed at step #4 (crates/formality-prove/src/prove/prove_wc.rs:87:14) because
                        judgment `prove_after { constraints: Constraints { env: Env { variables: [!const_0], coherence_mode: false }, known_true: true, substitution: {} }, goal: {@ Covers([value(0, bool)] , !const_0)}, assumptions: {@ ConstHasType(!const_0 , bool)}, decls: decls(222, [trait Foo <ty, const> where {@ ConstHasType(^const0_1 , bool)}, trait Bar <ty, ty, const> where {Foo(^ty0_1, const ^const0_2), @ ConstHasType(^const0_2 , bool)}], [impl Foo(u32, const value(0, bool)), impl <const> Bar(u32, u32, const ^const0_0) where {@ ConstHasType(^const0_0 , bool)}], [], [], [], [], {Bar, Foo}, {}) }` failed at the following rule(s):
                          the rule "prove_after" failed at step #1 (crates/formality-prove/src/prove/prove_after.rs:19:14) because
                            judgment `prove_wc_list { goal: {@ Covers([value(0, bool)] , !const_0)}, assumptions: {@ ConstHasType(!const_0 , bool)}, env: Env { variables: [!const_0], coherence_mode: false }, decls: decls(222, [trait Foo <ty, const> where {@ ConstHasType(^const0_1 , bool)}, trait Bar <ty, ty, const> where {Foo(^ty0_1, const ^const0_2), @ ConstHasType(^const0_2 , bool)}], [impl Foo(u32, const value(0, bool)), impl <const> Bar(u32, u32, const ^const0_0) where {@ ConstHasType(^const0_0 , bool)}], [], [], [], [], {Bar, Foo}, {}) }` failed at the following rule(s):
                              the rule "some" failed at step #0 (crates/formality-prove/src/prove/prove_wc_list.rs:28:14) because
                                judgment `prove_wc { goal: @ Covers([value(0, bool)] , !const_0), assumptions: {@ ConstHasType(!const_0 , bool)}, env: Env { variables: [!const_0], coherence_mode: false }, decls: decls(222, [trait Foo <ty, const> where {@ ConstHasType(^const0_1 , bool)}, trait Bar <ty, ty, const> where {Foo(^ty0_1, const ^const0_2), @ ConstHasType(^const0_2 , bool)}], [impl Foo(u32, const value(0, bool)), impl <const> Bar(u32, u32, const ^const0_0) where {@ ConstHasType(^const0_0 , bool)}], [], [], [], [], {Bar, Foo}, {}) }` failed at the following rule(s):
                                  the rule "exhaustive bool values cover variable" failed at step #4 (crates/formality-prove/src/prove/prove_wc.rs:163:17) because
                                    condition evaluted to false: `vals.len() == 2`
                      the rule "trait implied bound" failed at step #0 (crates/formality-prove/src/prove/prove_wc.rs:129:14) because
                        expression evaluated to an empty collection: `decls.trait_invariants()`
          the rule "trait implied bound" failed at step #0 (crates/formality-prove/src/prove/prove_wc.rs:129:14) because
            expression evaluated to an empty collection: `decls.trait_invariants()`
